File: 0257-binary-tree-paths/0257-binary-tree-paths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private void pathsfind(TreeNode root, String path, List<String>paths){
        if(root == null) return;

         if (!path.isEmpty()) {
            path += "->";
        }
        path += root.val;
        

         if (root.left == null && root.right==null) {
            paths.add(path);
        } else {
            pathsfind(root.left, path, paths);
            pathsfind(root.right, path, paths);
        }
    }
    public List<String> binaryTreePaths(TreeNode root) {
         List<String>paths = new ArrayList<>();
         pathsfind(root,"",paths);
         return paths;
    }
}

File: 0257-binary-tree-paths/0257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void pathsfind(TreeNode* root, string path, vector<string>&paths){
        if(!root) return;

         if (!path.empty()) {
            path += "->";
        }
        path += to_string(root->val);
        

         if (!root->left && !root->right) {
            paths.push_back(path);
        } else {
            pathsfind(root->left, path, paths);
            pathsfind(root->right, path, paths);
        }
    }
    vector<string> binaryTreePaths(TreeNode* root) {
         vector<string>paths;
         pathsfind(root,"",paths);
         return paths;
    }
};